### PR TITLE
Upgrade to new build 0.4.0

### PR DESCRIPTION
**Pull request by mechanomi**
this PR:

- includes a working GitHub Actions workflow for the docs
- removes the Travis CI tests (fixes https://github.com/crate/croud/pull/299)
- updates the Mergify configuration
- pins the docs build system to 0.4.0

see https://github.com/crate/crash/pull/341 for the origin of this work

see https://github.com/crate/tech-writing-domain/issues/353 for the parent issue

### CODE REVIEW

**Comment by mechanomi** [comment]
this should be changed to use the GitHub action authored by @mfussenegger on https://github.com/crate/crash/pull/341

**Comment by mechanomi** [comment]
fixes https://github.com/crate/croud/pull/299

**Comment by mechanomi** [comment]
can I please get a review from a one of the @crate/sysadmins?

additionally, before I can merge this PR, one of the @crate/sysadmins must update the branch protections for `master` like so:

- disable the Travis CI checks
- enable the new GitHub Actions `docs.yml` workflow

the changes made to the `.mergify.yml` can be used as a guide: https://github.com/crate/croud/pull/299/files#diff-6a30e47dd51449847422017b16878889b848f6602ac56d5a2e67ce8a19afade2

**Comment by mechanomi** [comment]
@chaudum could you please review this? ty!! <3

File: .gitignore [hunk 1]
```diff
@@ -77,4 +77,5 @@ env/
 docs/.utils
 docs/.clone
 docs/.crate-docs-build
```
Comment (chaudum) [inline .gitignore:79]: This line could be removed, right?

**Review by chaudum — CHANGES_REQUESTED** [review]
This PR does not remove the Travis configuration nor does it add the Github Actions workflows as stated in the description of the PR. Not sure if i

**Comment by mechanomi** [comment]
@chaudum how embarrassing! my push errored out and I didn't notice. I have rebased and pushed all the changes now, including a fixup that addresses the comment you left

**Review by chaudum — APPROVED** [review]

**Pull request merged**